tests/unit/transformers/test_base.py

```diff
@@ -1,10 +1,374 @@
+            'c': [7, 8, 9]
+        })
+        columns = 'a'
+        base_transformer = BaseTransformer()
+
+        # Run
+        base_transformer._store_columns(columns, data)
+        stored_columns = base_transformer.columns
+
+        # Assert
+        expected = ['a']
+        assert stored_columns == expected
+
+    def test__store_columns_missing(self):
+        """Test the ``_store_columns`` method when passed a missing column.
+
+        When the passed column does not exist in the dataframe, it should raise a ``KeyError``.
+
+        Input:
+            - a data frame.
+            - a list of column names, where at least one of the columns is not
+            present in the dataframe.
+
+        Raises:
+            - ``KeyError``, with the appropriate error message.
+        """
+        # Setup
+        data = pd.DataFrame()
+        columns = ['a', 'b']
+        base_transformer = BaseTransformer()
+        missing = set(columns)
+        error_msg = (f'Columns {missing} were not present in the data.')
+
+        # Run / Assert
+        with pytest.raises(KeyError, match=error_msg):
+            base_transformer._store_columns(columns, data)
+
+    def test__get_columns_data_multiple_columns(self):
+        """Test the ``_get_columns_data`` method.
+
+        The method should select the passed columns from the passed data.
+
+        Input:
+            - a dataframe.
+            - a list of a subset of the columns of the dataframe.
+
+        Output:
+            - the passed dataframe, but containing only the passed columns.
+        """
+        # Setup
+        data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4, 5, 6],
+            'c': [7, 8, 9]
+        })
+        columns = ['a', 'b']
+
+        # Run
+        columns_data = BaseTransformer._get_columns_data(data, columns)
+
+        # Assert
+        expected = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4, 5, 6],
+        })
+        pd.testing.assert_frame_equal(columns_data, expected)
+
+    def test__get_columns_data_single_column(self):
+        """Test the ``_get_columns_data`` method when passed a sigle column.
+
+        The method should select the passed column from the passed data, and convert it
+        into a pandas series.
+
+        Input:
+            - a dataframe.
+            - a list of one column from the dataframe.
+
+        Output:
+            - a pandas series, corresponding to the passed column from the dataframe.
+        """
+        # Setup
+        data = pd.DataFrame({
+            'a': [1, 2, 3],
+            'b': [4, 5, 6],
+            'c': [7, 8, 9]
+        })
+        columns = ['b']
+
+        # Run
+        columns_data = BaseTransformer._get_columns_data(data, columns)
+
+        # Assert
+        expected = pd.Series([4, 5, 6], name='b')
+        pd.testing.assert_series_equal(columns_data, expected)
+
     def test__set_columns_data_preserves_order_series(self):
```

Comment:
We can remove the `preserves_order` from this test name and the one below, since these are just test cases for `series` and `dataframe`

Reply:
We should also add 2 test cases that pass 1d and 2d numpy arrays